charts/astronomer/templates/_helpers.yaml

```diff
@@ -41,6 +41,37 @@
 {{- end }}
 {{- end }}
 
+{{/*
+Returns a repository prefix to be used to construct image names throughout
+the platform. Defaults to quay.io/astronomer but uses the value defined in
+global.privateRegistry.repository when enabled.
+*/}}
+{{ define "astronomer.platform_registry" -}}
+{{- if .Values.global.privateRegistry.enabled -}}
+{{ .Values.global.privateRegistry.repository }}
+{{- else -}}
+quay.io/astronomer
+{{- end }}
+{{- end }}
```

Comment:
Is this needed? The only place `astronomer.platform_registry` is used is `authSidecar.repository`.  I'm still having a hard time understanding why all of this cannot be the same as, say, `registry.image`. EG, structured in a way where the following big works:

```
{{ define "authSidecar.image" -}}
{{- if .Values.global.authSidecar.enabled -}}
{{ .Values.global.privateRegistry.repository }}/ap-auth-sidecar:{{ .Values.images.registry.tag }}
{{- else -}}
{{ .Values.images.registry.repository }}:{{ .Values.images.registry.tag }}
 {{- end }}
 {{- end }}
```